app/helpers/patient_query_helper.rb

```diff
@@ -507,23 +507,35 @@ def advanced_filter_lab_result(patients, filter)
     filter[:value].each do |field|
       case field[:name]
       when 'lab-type'
-        labs = labs.where('laboratories.lab_type = ?', field[:value])
+        labs = if filter[:value].blank?
```

Comment:
can you update the demo script to sometimes create labs with fields that are blank here? https://github.com/SaraAlert/SaraAlert/blob/master/lib/tasks/demo.rake#L679

can probably do:
```
laboratory = Laboratory.new(
        patient_id: patient_id,
        lab_type: ['PCR', 'Antigen', 'Total Antibody', 'IgG Antibody', 'IgM Antibody', 'IgA Antibody', 'Other', ''].sample,
        created_at: lab_ts,
        updated_at: lab_ts
      )
laboratory[:specimen_collection] = create_fake_timestamp(beginning_of_day - 1.week, beginning_of_day) if rand < 0.95
laboratory[:report] = create_fake_timestamp(beginning_of_day) if rand < 0.95
laboratory[:result] = result if rand < 0.95
laboratories << laboratory
```

Reply:
done https://github.com/SaraAlert/SaraAlert/pull/961/commits/ca81af3cc9ddd1092d5072551eaa1e378a0d0fd4